vcr/holder/interface.go

```diff
@@ -47,6 +48,9 @@ type Wallet interface {
 	// If signerDID is not provided, it will be derived from the credentials credentialSubject.id fields. But only if all provided credentials have the same (singular) credentialSubject.id field.
 	BuildPresentation(ctx context.Context, credentials []vc.VerifiableCredential, options PresentationOptions, signerDID *did.DID, validateVC bool) (*vc.VerifiablePresentation, error)
 
+	// BuildSubmission builds a Verifiable Presentation based on the given presentation definition.
+	BuildSubmission(ctx context.Context, walletDID did.DID, presentationDefinition pe.PresentationDefinition, acceptedFormats map[string]map[string][]string, params BuildParams) (*vc.VerifiablePresentation, *pe.PresentationSubmission, error)
```

Comment:
should we add a `type Formats map[string]map[string][]string` in the `pe` package? This is an incomprehensible data format that is used 30+ times of which half in non-test code.

Reply:
correct, but let's do that separately to prevent merge conflicts.